### DISCUSSION TITLE
Alguém pode me dizer o que há de errado com meu código?

### DISCUSSION THREAD

**Discussion by felipethecreator** [community/community#68521] [Programming Help] [answered]
### Body

function criarFuncoes() {
  var funcoes = [];
  for (var i = 0; i < 5; i++) {
    funcoes.push(function() {
      console.log(i);
    });
  }
  return funcoes;
}

var minhasFuncoes = criarFuncoes();

minhasFuncoes[0]();
minhasFuncoes[1]();
minhasFuncoes[2]();

### Guidelines

- [x] I have read and understood this category's [guidelines](https://github.com/orgs/community/discussions/51384) before making this post.

**Comment by Marichoii** [comment, marked as answer]
No seu caso, todas as funções dentro do array "funcoes" estão fazendo referência à mesma variável i, que, após o término do loop, tem o valor 5, você chama minhasFuncoes0, minhasFuncoes1 e minhasFuncoes2, todas elas imprimirão o valor 5, em vez dos valores 0, 1 e 2 que você pode estar esperando.

function criarFuncoes() {
  var funcoes = [];
  for (var i = 0; i < 5; i++) {
    (function(j) {
      funcoes.push(function() {
        console.log(j);
      });
    })(i);
  }
  return funcoes;
}

var minhasFuncoes = criarFuncoes();

minhasFuncoes[0]();
minhasFuncoes[1](); 
minhasFuncoes[2](); 

Faça desse jeito e irá funcionar!

**Reply by felipethecreator** [reply]
Muito obrigado!!
